app/app.go

```diff
@@ -88,7 +88,12 @@ func (app *app) Setup(opts Options) error {
 		return bosherr.WrapError(err, "Getting platform")
 	}
 
-	settingsSourceFactory := boshinf.NewSettingsSourceFactory(config.Infrastructure.Settings, app.platform, app.logger)
+	settingsSourceFactory := boshinf.NewSettingsSourceFactory(
+		app.platform.GetFs(),
+		filepath.Join(app.dirProvider.BoshDir(), "httpRegistryAccessCache"),
+		config.Infrastructure.Settings,
+		app.platform,
+		app.logger)
```

Comment:
Hi James @jfmyers9 ,

Thanks for reviewing our PR. We ever discussed a lot about this issue with Dmitriy before, let me write down the background there. The problem is specific to IBM Cloud Infrastructure (previously known as Softlayer) and especially for our huge cloud foundry production environments, we have thousands of VM deployed under one Softlayer account, but Softlayer has a rate limit policy that there only allows 50 requests sending to SL api endpoint per second in one account. So you can imagine if the BOSH director is down (like for upgrade), bosh agent on thousands of VM will restart almost at the same time, they will send request to SL metadata server for retrieving http registry access info, it will be far beyond rate limit, and since bosh agent will continuously try to restart itself until it gets what it wants, which will get the situation worse, we ever got complaint from SL that we made their endpoint server and UI slower. This is the issue we want to address by this PR. So rather than retrieving http registry access info from SL metadata server, we want to cache this information in local (we can't use memory cache since we need to keep this cache constantly even if the system is reboot). The real user data is still stored in http registry on bosh director, (actually, we wanted to use SL metadata server to store user data, but we have to give up also due to rate limit issue.), I don't think we need to cache user data in local since we can retrieve it from director at any time without hitting rate limit issue. 

I agree with your concern about the cache invalidation, we don't figure out a good way to address it, but from our practice, we generally don't often change url/credentials of director http registry, the cache will be refreshed when vm recreation. We will add more UT there, thanks for pointing it out.

Please let us know if you still have questions/concerns. Thanks.

/cc @jrussett @edwardstudy